src/components/NewPoll/NewPollForm.js

```diff
@@ -1,101 +1,68 @@
-    dispatch(actions.updateQuestion(newQuestion));
-
-  // using useCallback from react to wrap useDispatch of react-redux , useCallback is an overkill for now , but if we are to pass updateAnswers to a child component it is better to wrap it as given to prevent child component rerendering
-  const updateAnswers = newAnswer =>
-    dispatch(actions.updateResponses(newAnswer));
-
-  const updateChannel = newChannel =>
-    dispatch(actions.updateChannel(newChannel));
-
-  const updateChannelID = newChannelID =>
-    dispatch(actions.updateChannelID(newChannelID));
-
-  const updateChannelSize = newChannelSize =>
-    dispatch(actions.updateChannelSize(newChannelSize));
-
-  const handleSubmitPoll = event => {
-    event.preventDefault();
-
-    let isResponseValid = responses.length >= 2;
-    responses.forEach(response => {
-      if (isResponseValid && response == "") {
-        isResponseValid = false;
-      }
-    });
-
-    // setState();
-    if (question == "" || !isResponseValid || channel == "") {
-      // but you can use a location instead
-      setError(true);
-      console.log("Make an error message for each box below");
-    } else {
-      setError(false);
-      history.push("/poll-submitted");
-    }
+import Response from "./Response";
+
+const NewPollForm = ({
+  updateQuestion,
+  updateAnswers,
+  updateChannel,
+  updateChannelID,
+  updateChannelSize,
+  message,
+  fetchChannels,
+  channels
+}) => {
+  useEffect(() => {
+    fetchChannels();
+  }, [fetchChannels]);
+
+  const handleSubmitPoll = e => {
+    e.preventDefault();
   };
 
-  useEffect(() => {
-    fetch(API_BASE_URL + "/channels")
-      .then(res => res.json())
-      .then(data => {
-        const _data = [
-          {
-            name: "Select a channel",
-            id: 0
-          }
-        ];
-        return _data.concat(data);
-      })
-      .then(data => setChannels(data))
-      .catch(error => {
-        console.log("error: ", error);
-      });
-  }, []);
-
-  const deleteResponse = (event, index) => {
-    event.preventDefault();
-    const _responses = [...responses];
-    _responses.splice(index, 1);
-    // setResponses(_responses);
+  const updateResponse = e => {
+    /* 
+      The input id's are unique and are in the format 'response{indexNumber}', using regEx to grab the {indexNumber}
+
+      We should probably update this function later so that redux handles updating the array.     
+    */
+    const idx = numberParser(e.target.id);
+    const _responses = [...message.responses];
+    _responses[idx] = e.target.value;
     updateAnswers(_responses);
   };
 
-  const handleResponseChange = event => {
-    const _responses = [...responses];
-    _responses[event.target.dataset.id] = event.target.value;
-    // setResponses(_responses);
+  const deleteResponse = e => {
+    // Again, we should probably let redux handle updating the array.
+    const idx = parseInt(e.target.attributes.value.value);
```

Comment:
same comment about passing in the index here

Reply:
updated here.